backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -268,7 +278,11 @@ public Endpoint getEndpoint(@Context SecurityContext sec, @PathParam("id") UUID
         } else {
             // Fetch the secrets from Sources.
             if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
-                this.secretUtils.loadSecretsForEndpoint(endpoint);
+                try {
+                    this.secretUtils.loadSecretsForEndpoint(endpoint);
+                } catch (final SourcesException e) {
+                    Log.error(e.getMessage(), e);
+                }
```

Comment:
Here, we're loading a single endpoint and an exception is thrown while retrieving its secret from Sources. Is it wise returning inconsistent data or should we show the caller that something went wrong?

Reply:
I am not sure... in this case I wouldn't? Because the user might want to edit or view what the integration has, and if we're not able to show it fully... it might be better to simply throw an error in here?

Reply:
Could we  have more information about the `SourcesException`? If it is a timeout exception or something that could succed after a retry, then it's worth to fail the request.

IF it's something like "secret not found" it might be worth to let the user know that we couldn't retrieve the secret and allow the user to update to something new.